### PR TITLE
Add Evergreen setup

### PR DESCRIPTION
**Pull request by louisaberger**
@TomerYakir -- the mongos int test is still failing, but this should put the structure in place!

### CODE REVIEW

File: server_test.go [hunk 1]
```diff
@@ -379,49 +379,51 @@ func checkClient(opts *options.ClientOptions) error {
...
+// 	opts := options.Client().ApplyURI(fmt.Sprintf("mongodb://127.0.0.1:%d", port))
+// 	for i := 0; i < 10; i++ {
+// 		if err := checkClient(opts); err != nil {
+// 			t.Error(err)
+// 		}
+// 	}
+// 	sessCtrCurr := atomic.LoadInt32(&sessCtr)
+
+// 	if sessCtrCurr != int32(20) {
+// 		t.Errorf("expect session counter to be 20 but got %d", sessCtrCurr)
```
Comment (TomerYakir) [inline server_test.go:420]: fwiw this is fixed by changing the counter to be 30 instead of 20. I fixed that on #61.

**Review by TomerYakir — CHANGES_REQUESTED** [review]
Looks like there's a merge conflict. Otherwise LGTM!

File: server_test.go (hunk 1 above)
Comment (louisaberger) [reply server_test.go:420]: Thank you! I'm going to merge in with commented-out, would you mind uncommenting in 61?

Comment (TomerYakir) [reply server_test.go:420]: sure!

**Pull request merged**